File: Programming Languages/c++ language/Practice Questions/factorial.cpp
#include <iostream>

using namespace std;

int factorial(int n)
{
    int fact = 1;

    for (int i = 1; i <= n; i++)
    {
        fact *= i;
    }
    return fact;
}

int main()
{
    int n;
    cout << "Enter number : ";
    cin >> n;
    cout << "factorial = " << factorial(n);
    return 0;
}

File: Programming Languages/Java/7. Methods/Recursions/Factorial.java
public class Factorial {
    static int factorial(int n){
        if (n==0 || n==1) {
            return 1;
        }
        // recursion
        return n * factorial(n-1);
    }
    public static void main(String[] args) {
        System.out.println(factorial(6));
    }
}
